doc/Programming-guide/Your-first-Kaa-application/attach/demo-cpp/KaaDemo.cpp

```diff
@@ -0,0 +1,132 @@
+/**
+ *  Copyright 2014-2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <memory>
+#include <cstdint>
+#include <string>
+#include <vector>
+#include <list>
+#include <boost/asio.hpp>
+#include <kaa/Kaa.hpp>
+#include <kaa/IKaaClient.hpp>
+#include <kaa/configuration/manager/IConfigurationReceiver.hpp>
+#include <kaa/configuration/storage/FileConfigurationStorage.hpp>
+#include <kaa/log/strategies/RecordCountLogUploadStrategy.hpp>
+
+using namespace kaa;
+ 
+ 
+
+class Transceiver : public IConfigurationReceiver {
+private:
+	// Static variables 
+	static const std::string savedConfig;
+	int32_t samplePeriod;
+	std::shared_ptr<IKaaClient> kaaClient;
+	boost::asio::io_service service;
+	boost::posix_time::seconds interval;
```

Comment:
Trailing underscore is needed:
https://google.github.io/styleguide/cppguide.html#Variable_Names